### PR TITLE
Add Feather STM32F405 Express

### PR DESCRIPTION
**Pull request by hierophect**
This PR adds support to the Feather F405 board from Adafruit.

### CODE REVIEW

**Comment by hierophect** [comment]
I'm not familiar with Shippable. Is there something I still need to change? I cannot reproduce the problem it is having locally, and it doesn't provide very useful error information.

**Comment by hierophect** [comment]
The specific error provided by Shippable is:
`error: ARM (defined at arch/Kconfig:26) is assigned in a configuration file, but is not directly
user-configurable (has no prompt). It gets its value indirectly from other symbols. See
http://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_ARM.html and/or look up ARM in the
menuconfig/guiconfig interface. The Application Development Primer, Setting Configuration Values,
and Kconfig - Tips and Best Practices sections of the manual might be helpful too.`

This implies CONFIG_ARM is not user settable. However, it is set in every other ST board file listed in the device tree - I'm not sure what is different about my PR, so I'd appreciate anyone's help in spotting the error!

File: boards/arm/adafruit_feather_stm32f405/adafruit_feather_stm32f405_defconfig [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_ARM=y
+CONFIG_BOARD_ADAFRUIT_FEATHER_STM32F405=y
```
Comment (erwango) [inline boards/arm/adafruit_feather_stm32f405/adafruit_feather_stm32f405_defconfig]: Both these lines should be removed (recent evolution as well).

File: boards/arm/adafruit_feather_stm32f405/adafruit_feather_stm32f405_defconfig [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+CONFIG_CLOCK_STM32_PLL_M_DIVISOR=12
+CONFIG_CLOCK_STM32_PLL_N_MULTIPLIER=336
+CONFIG_CLOCK_STM32_PLL_P_DIVISOR=2
+CONFIG_CLOCK_STM32_PLL_Q_DIVISOR=7
+CONFIG_CLOCK_STM32_AHB_PRESCALER=1
+CONFIG_CLOCK_STM32_APB1_PRESCALER=4
+CONFIG_CLOCK_STM32_APB2_PRESCALER=2
+
+#enable pwm
+CONFIG_PWM=y
```
Comment (erwango) [inline boards/arm/adafruit_feather_stm32f405/adafruit_feather_stm32f405_defconfig]: Aside from minimum required for basic operations (GPIO, clock, UART), no peripheral should be enabled in default board configuration.

File: boards/arm/adafruit_feather_stm32f405/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+The STM32F405 SPI2 peripheral is available on the SCK (PB13), MI
+(PB14) and MO (PB15) pins.
+
+Programming and Debugging
+*************************
+
+Only JLink programming is supported through Zephyr at this time. Use
+of the built in DFU-Util bootloader is possible by following the
+instructions on the Learn website.
```
Comment (erwango) [inline boards/arm/adafruit_feather_stm32f405/doc/index.rst]: Maybe you can use the link here as well.

File: boards/arm/adafruit_feather_stm32f405/pinmux.c [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+#endif /* CONFIG_I2C_1 */
+#ifdef CONFIG_SPI_2
+	{STM32_PIN_PB13, STM32F4_PINMUX_FUNC_PB13_SPI2_MASTER_SCK},
+	{STM32_PIN_PB14, STM32F4_PINMUX_FUNC_PB14_SPI2_MASTER_MISO},
+	{STM32_PIN_PB15, STM32F4_PINMUX_FUNC_PB15_SPI2_MASTER_MOSI},
+#endif /* CONFIG_SPI_2 */
+#ifdef CONFIG_USB_DC_STM32
+	{STM32_PIN_PA11, STM32F4_PINMUX_FUNC_PA11_OTG_FS_DM},
+	{STM32_PIN_PA12, STM32F4_PINMUX_FUNC_PA12_OTG_FS_DP},
+#endif	/* CONFIG_USB_DC_STM32 */
```
Comment (erwango) [inline boards/arm/adafruit_feather_stm32f405/pinmux.c]: Please fix comment alignment

File: boards/arm/adafruit_feather_stm32f405/pinmux.c [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+{
+	ARG_UNUSED(port);
+
+	stm32_setup_pins(pinconf, ARRAY_SIZE(pinconf));
+
+	return 0;
+}
+
+SYS_INIT(pinmux_stm32_init, PRE_KERNEL_1,
+		CONFIG_PINMUX_STM32_DEVICE_INITIALIZATION_PRIORITY);
```
Comment (erwango) [inline boards/arm/adafruit_feather_stm32f405/pinmux.c]: Please fix indentation

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this addition!
A few hints and nitpicks otherwise LGTM.

**Comment by hierophect** [comment]
Botched my git here by accidentally force pushing without committing, which forces future commits to be merges. Do I need to nuke this PR and start over? What's your best practice here?

**Comment by hierophect** [comment]
Something's gone pretty wrong, Github won't even show my latest changes any more. Nuking and starting over.

**Comment by hierophect** [comment]
@erwango please see #23080

**Comment by erwango** [comment]
@hierophect ok, will do. But please keep "close PR"/"create new PR" exceptional. This is much harder to track changes happening between reviews.

**Comment by hierophect** [comment]
@erwango sorry, I'm not used to the single-commit approach and got myself into a corner I wasn't sure how to git-fu myself out of. I'll do some research and see how I can fix it properly next time.

**Pull request closed without merging**